test/battle/hold_effect/metronome.c

```diff
@@ -33,8 +37,117 @@ SINGLE_BATTLE_TEST("Metronome gradually boosts power by 20%")
         }
     }
 }
+
+SINGLE_BATTLE_TEST("Metronome Item's boost is reset if the attacker uses a different move")
+{
+    s16 damage[2];
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_METRONOME); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_TACKLE); }
+        TURN { MOVE(player, MOVE_QUICK_ATTACK); }
+        TURN { MOVE(player, MOVE_TACKLE); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        HP_BAR(opponent, captureDamage: &damage[0]);
+
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_QUICK_ATTACK, player);
+
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        HP_BAR(opponent, captureDamage: &damage[1]);
+    } THEN {
+        EXPECT_EQ(damage[0], damage[1]);
+    }
+}
+
+SINGLE_BATTLE_TEST("Metronome Item's boost is reset if the move fails")
+{
+    s16 damage[2];
+    KNOWN_FAILING; //https://github.com/rh-hideout/pokeemerald-expansion/issues/3251
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_METRONOME); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_TACKLE); }
+        TURN { MOVE(opponent, MOVE_PROTECT); MOVE(player, MOVE_TACKLE); }
+        TURN { MOVE(player, MOVE_TACKLE); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        HP_BAR(opponent, captureDamage: &damage[0]);
+
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        HP_BAR(opponent, captureDamage: &damage[1]);
+    } THEN {
+        EXPECT_EQ(damage[0], damage[1]);
+    }
+}
+
+SINGLE_BATTLE_TEST("Metronome Item counts called moves instead of the calling move")
+{
+    s16 damage[2];
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_METRONOME); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_METRONOME, WITH_RNG(RNG_METRONOME, MOVE_TACKLE)); }
+        TURN { MOVE(player, MOVE_METRONOME, WITH_RNG(RNG_METRONOME, MOVE_TACKLE)); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        HP_BAR(opponent, captureDamage: &damage[0]);
+
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        HP_BAR(opponent, captureDamage: &damage[1]);
+    } THEN {
+        EXPECT_MUL_EQ(damage[0], UQ_4_12(1.2), damage[1]);
+    }
+}
+
+SINGLE_BATTLE_TEST("Metronome Item counts charging turn of moves for its attacking turn", s16 damage)
+{
+    u32 item;
+
+    KNOWN_FAILING; // https://github.com/rh-hideout/pokeemerald-expansion/issues/3250
+    PARAMETRIZE {item = ITEM_NONE; }
+    PARAMETRIZE {item = ITEM_METRONOME; }
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_SOLAR_BEAM].effect == EFFECT_SOLAR_BEAM);
+        PLAYER(SPECIES_WOBBUFFET) { Item(item); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_SOLAR_BEAM); }
+        TURN { SKIP_TURN(player); }
+    } SCENE {
+        HP_BAR(opponent, captureDamage: &results[i].damage);
```

Comment:
How is it playing locally? I'm thinking maybe the test is capturing the "damage" of the charging turn, causing it to fail.

Reply:
I added more verbose `MESSAGE` statements so it is clear that `HP_BAR` captures the damage turn. :)

If you look at the actual error message you can see that the damage is not `0`:
```
test/battle/hold_effect/metronome.c:131: EXPECT_MUL_EQ(51, 1.19:, 51) failed: 61 not in [50..52]
```
too :)

Reply:
Awesome, thank you!